src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -92,6 +92,42 @@
       :else
       false)))
 
+(defn validate-regex-backreferences!
+  "Validates that allow and deny back-references actually exist as capture
+   groups in the path regex. Throws an exception if a back-reference refers
+   to a capture group that doesn't exist."
+  [rule]
+  (let [group-count (try
+                      (-> rule
+                          :match-request
+                          :path
+                          re-pattern
+                          (.matcher "")
+                          .groupCount)
+                      (catch PatternSyntaxException e
```

Comment:
This was originally around the whole function body, but only these first two lines could throw this.
I moved the catch up around only those lines, and then combined the two lines into a `->`.

Happy to reformat this any way, mainly just wanted to reduce the scope of the try/catch